packages/sdk/snet/sdk/service_client.py

```diff
@@ -129,4 +153,6 @@ def deposit_and_open_channel(self, amount, expiration):
 
     def _get_newly_opened_channel(self, receipt):
         open_channels = self.sdk.mpe_contract.get_past_open_channels(self.sdk.account, self, receipt["blockNumber"], receipt["blockNumber"])
+        if len(open_channels) == 0:
```

Comment:
Lets check whats done the snet-cli code. We may need to wait and retry a few times before we error out

Reply:
Looks like the behaviour is the same in snet-cli as well. Resolve for now but we need to track this.